src/Microsoft.AspNet.Server.Kestrel/KestrelServerInformation.cs

```diff
@@ -22,10 +22,7 @@ public class KestrelServerInformation : IKestrelServerInformation, IServerAddres
         public void Initialize(IConfiguration configuration)
         {
             var urls = configuration["server.urls"] ?? string.Empty;
-            foreach (var url in urls.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries))
-            {
-                Addresses.Add(url);
-            }
+            Addresses = urls.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
I think this breaks here:

https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs#L262

We don't want the collection to be read only.